src/amo/withExperiment.js

```diff
@@ -302,6 +312,25 @@ export const withExperiment =
         return Object.keys(registeredExperiments).includes(id);
       }
 
+      // Warning: Please think carefully before using this function. It was
+      // introduced to address a specific requirement for an experiment to
+      // avoid having to store data in a cookie, but realistically one should
+      // never change the variant for a user after they have been enrolled in
+      // an experiment.
+      // See https://github.com/mozilla/addons-frontend/issues/10770
+      updateVariant = (variant: string) => {
+        const registeredExperiments = this.getExperiments();
+        invariant(
+          this.cookieIncludesExperiment(registeredExperiments),
+          'Trying to update a variant for an experiment that is not in the cookie',
+        );
+
+        const experimentsToStore = { ...registeredExperiments };
+        experimentsToStore[id] = variant;
+
+        this.writeCookie(experimentsToStore);
```

Comment:
One problem with this is that it will change the expiration date of the existing cookie.

Reply:
It should not be a problem. You are correct that a new user enrolled on the same day will also have a cookie for 30 days. Also, we remove the experiment from the cookie as soon as it has been disabled, so the expiration date being extended isn't really an issue.